### PR TITLE
softrend: linux support

### PR DESCRIPTION
**Pull request by skmp**
reicast-linux-softrend

The current softrend is x86 specific due to heavy simd use. It's also windows specific because I was always lazy as always. This is some WIP to get it working under linux.

Current state
- Fixed intrin code for linux (x64)
- Ifdef'd out windows rendering code
- Compiles and runs and draws w/o crashing
- Presents to screen via dirty X11 code
- Makefile changes for x86, x64 (sse4.1, openmp)
- Depends on sse4.1
- Makefile flag HAS_SOFTREND, needs sse4.1. Automatically set via /proc/cpuinfo for x86/x64 targets (clearly not ideal when compiling for distribution, though)

Todo (for this PR)
- Fix windows
- Improve makefile for distribution / etc?
 - This is gonna be a pain...

Wishlist (maybe for this PR)
- Some kind of abstraction for pushing buffers to screen

Further plans
- Re-work a bit the 3d stack so that it's initialized in a more sane way [general graphics related]
- Investigate sse2neon for android/neon support
- Refine the the code into a "reference rasterizer" and a "fast rasterizer"
- Aim for code cleanness and pixel accuracy refrast
- Aim for portable, multi-threaded, simd, possibly jitting fastrast
 - And pixel accurate, ofc

### CODE REVIEW

File: core/linux-dist/evdev.h [hunk 1]
```diff
@@ -5,37 +5,37 @@
 struct EvdevControllerMapping
```
Comment (skmp) [inline core/linux-dist/evdev.h]: Huh, this looks unrelated

File: core/rec-cpp/rec_cpp.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #include "hw/sh4/dyna/shil_canonical.h"
 
 
-#define MIPS_COUNTER 0
```
Comment (skmp) [inline core/rec-cpp/rec_cpp.cpp]: This is unrelated, too

**Comment by skmp** [comment]
Rebased to master, removed accidental changes!

**Comment by skmp** [comment]
@Holzhaus what's your take on makefile rules for this? I really hate how it depends on sse4.1 -- and enabling it globally makes even non-softrend paths depend on it

**Comment by Holzhaus** [comment]
@skmp Looks good to me (almost). One minor change: [I'd move the autodetection of SSE4.1 to the location where `$(platform)` is set.](https://github.com/Holzhaus/reicast-emulator/commit/f8672bd1b04aaaeae87f7c2637ddd6cc32378002) That way, a user can still override autodetection by defining the platform variable on the command line:

```
make -C shell/linux platform="x86 sse4_1"
```

**Comment by skmp** [comment]
Squashed your changes and also fixed windows. I thought of doing it that way but lazyness got the better of me

**Comment by Shoegzer** [comment]
Hello, I had issues compiling the latest.. a git bisect revealed this commit as the source.  Specifically, when doing a make, I get:

Platform: x64 sse4_1
Makefile:319: warning: overriding commands for target `obj-dreamcast-x64'
Makefile:315: warning: ignoring old commands for target `obj-dreamcast-x64'
Makefile:323: warning: overriding commands for target `obj-dreamcast-x64'
Makefile:319: warning: ignoring old commands for target `obj-dreamcast-x64'
make: *** No rule to make target `../../core/%.S', needed by `obj-dreamcast-x64'.  Stop.
